### DISCUSSION TITLE
Danfos Ally will not update to latest ota firmware

### DISCUSSION THREAD

**Discussion by WouterKr** [Koenkk/zigbee2mqtt#12275] [Devices]
Hi,

I am not able to update two Danfoss Ally thermostates to the latest ota firmware. 
One is running v01.02.0008 01.02 and the other one v01.12.0008 01.12.
As I understand it both should upgrade to 1.18 which is available, but the logging is showing:

` Received Zigbee message from 'Bedroom1 - RadiatorThermostaat', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":274,"imageType":256,"manufacturerCode":4678}' from endpoint 1 with groupID null
Debug Device 'Bedroom1 - RadiatorThermostaat' requested OTA
Debug Got OTA request '{"fieldControl":0,"manufacturerCode":4678,"imageType":256,"fileVersion":274}'
Debug ZigbeeOTA: downloaded main index
Debug Is new image available for '0x842e14fffe2ab972', current '{"fieldControl":0,"manufacturerCode":4678,"imageType":256,"fileVersion":274}', latest meta '{"fileVersion":274,"fileSize":379806,"url":"https://github.com/Koenkk/zigbee-OTA/raw/master/images/Danfoss/1246-0100-01180118.0002_%28F5BFD8B8%29.ota","sha512":"873effaa113e1d07643fbf25f6e26ee4b23f8deb7d006cce7fc490fd8ad7a5754de4c352b6f3d5e172022a34ea2699f5d1d8fda90f4348b25d5feaa169e38c85"}'
Debug Update available for '0x842e14fffe2ab972': NO
Info No update available for 'Bedroom1 - RadiatorThermostaat'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"No update available for 'Bedroom1 - RadiatorThermostaat'","meta":{"device":"Bedroom1 - RadiatorThermostaat","status":"not_available"},"type":"ota_update"}'
Info MQTT publish: topic 'zigbee2mqtt/Bedroom1 - RadiatorThermostaat', payload '{"adaptation_run_control":"None","adaptation_run_settings":false,"adaptation_run_status":null,"algorithm_scale_factor":1,"battery":63,"day_of_week":"saturday","external_measured_room_sensor":1982,"heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":255,"load_balancing_enable":true,"load_estimate":-8000,"load_room_mean":-8000,"local_temperature":19.34,"mounted_mode_active":false,"mounted_mode_control":false,"occupied_heating_setpoint":18,"pi_heating_demand":0,"preheat_status":false,"programming_operation_mode":"setpoint","radiator_covered":false,"regulation_setpoint_offset":0,"running_state":"idle","setpoint_change_source":"externally","system_mode":"heat","thermostat_vertical_orientation":false,"trigger_time":660,"update":{"state":"idle"},"update_available":false,"viewing_direction":false,"window_open_external":false,"window_open_feature":true,"window_open_internal":"closed"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"Bedroom1 - RadiatorThermostaat","updateAvailable":false},"status":"ok","transaction":"jbpz2-2"}'
`

I can not figure out what is wrong but hope someone can help me.
Thanks in advance.

Regards,
Wouter

**Comment by lux73** [comment]
there are several Updates with last z2m Updates for Danfoss Ally Device

I would recommend to delete and re-pair one of them and start the Update via z2m Webinterface

i have updated both of mine without problem but have to delete and re-pair after updating before it shows the correct updated Firmware. Mine had 1.12 installed before.

**Comment by WouterKr** [comment]
Thanks for the tip. 

Unfortunatly it did not help. For some reason the one with v01.02.0008 01.02 shows now 01.18.0008 01.18 with a build date 20201119 which seems not right. The other one has not changed. The update on both the valves is not possible e.g. not available.
